NAS2D/Resource/AnimationSet.h

```diff
@@ -42,9 +42,9 @@ namespace NAS2D
 		const std::vector<Frame>& frames(const std::string& actionName) const;
 
 	private:
-		std::string mFileName;
-		std::map<std::string, std::string> mImageSheetMap;
-		std::map<std::string, std::vector<Frame>> mActions;
+		std::string mFileName{};
+		std::map<std::string, std::string> mImageSheetMap{};
+		std::map<std::string, std::vector<Frame>> mActions{};
```

Comment:
I'm hesitant to set default initial values here. They have a tendency to silence warnings for uninitialized values when you may actually want those warnings. In this particular class, you might reasonably expect the constructor to explicitly set all fields, and should get a warning if a field is not explicitly set.

Somewhat less relevant, though for certain classes "empty" values might not even be allowed. That's not a problem here for the top level collections, though the nested frame list is not allowed to be empty.

Reply:
The constructors do initialize the members. If not explicitly modified via the initializer list, the default value is used. In the specific case of the offending constructor, the `mImageSheetMap` and `mActions` are initialized in the body of the constructor at the bottom so they are "initialed to defaults" only for a brief period prior to the body being entered.

Reply:
I feel like you're missing the point here.

If you comment out one of those constructor lines that properly initializes one of those variables, you will no longer get a warning, since it's now default initialized.

Adding default initializers inhibits the compiler's ability to check for errors. It shouldn't be done unless the defaults are real meaningful defaults, and there are code paths that would make use of them.